include/NeoFOAM/fields/segmentedField.hpp

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include "NeoFOAM/core/primitives/label.hpp"
+#include "NeoFOAM/fields/field.hpp"
+
+namespace NeoFOAM
+{
+
+/**
+ * @brief Compute the segments from intervals by accumulating the intervals.
+ *
+ * The segments are computed by accumulating the intervals. So, with given
+ * intervals are {1, 2, 3, 4, 5} would result in the segments {0, 1, 3, 6, 10, 15}.
+ * Note that the segments are one element longer than the intervals
+ * and are assumed to be zero
+ *
+ * @param intervals The intervals to compute the segments from.
+ * @param intSpan The span of intervals.
+ * @param segSpan The span to store the segments in.
```

Comment:
Ok thanks with the docstring it is much easier to understand now. I would change the wording a bit, since it is a bit confusing (at least to me). I would replace [interval](https://en.wikipedia.org/wiki/Interval_(mathematics) with `size` and segments with `offsets`
see also https://en.wikipedia.org/wiki/Prefix_sum

Additionally, I think we can clean the function up by just passing in fields and get the spans in the function body. Then we have one function argument less and we can make sure that intervals and intSpan are consistent.

```suggestion
 * @brief Compute segment offsets from an input field corresponding to lenghts by computing a prefix sum.
 *
 * The offsets are computed by a prefix sum of the input values. So, with given
 * input of {1, 2, 3, 4, 5} the offets are {0, 1, 3, 6, 10, 15}.
 * Note that the length of offSpan must be  length of valSpan + 1 
 * and are all elements of offSpan are required to be zero
 *
 * @param[in] in The values to compute the offsets from.
 * @param[in,out] offsets The field to store the resulting offsets in. 
```